src/libs/OptionsListUtils.js

```diff
@@ -775,13 +806,29 @@ function getCurrencyListForSections(currencyOptions, searchValue) {
  *
  * @param {Object} report
  * @param {Object} personalDetails
- * @returns {String}
+ * @returns {Array<String>}
  */
 function getReportIcons(report, personalDetails) {
     // Default rooms have a specific avatar so we can return any non-empty array
     if (ReportUtils.isChatRoom(report)) {
         return [''];
```

Comment:
I know this wasn't added here, but anyone know what is up with the returning an array with an empty string? What is that supposed to mean and why should it be "non-empty"?

Reply:
I believe we return an empty string because we're using an Icon component instead of using a url for an Image when we're rendering the Avatars for rooms. This could be null, or empty or I guess we could pass the Icon component itself here, but the way we built it just has [RoomAvatar.js](https://github.com/Expensify/App/blob/33e5d99b95e8e7f767c42f5dccbb76879a565387/src/components/RoomAvatar.js#L20) just ignore whatevers stored here anyway. I think having this be empty string satisfies some propType checks.

Reply:
Ok thanks for the explanation. Maybe we can return an empty array or allow the icons to be `undefined` and not have to say anything about how the report icons will be used or why the array must be non empty. But don't think we have to block this PR on it though - just something I was curious about.

Reply:
@marcochavezf Did our changes to `Avatar` change this?

Reply:
Ah no, this `getReportIcons` is called before we save reports into Onyx and we can't set the corresponding Icon components here, as I suggested in this [comment](https://github.com/Expensify/App/pull/7852#discussion_r817152760) I think (in order to move/merge the logic of `getAvatarSources` to `getReportIcons`) we'll need to implement a different approach or maybe remove `getReportIcons` from `PersonalDetails`: 

https://github.com/Expensify/App/blob/main/src/libs/actions/PersonalDetails.js#L192-L211 

But I can address it in another ticket if it's ok because my gut tells me this can break something and we'll require more time to address it :D